components/validator/validator.go

```diff
@@ -132,6 +140,12 @@ func (v *Validator) Start() error {
 		return fmt.Errorf("cannot start challenger: %w", err)
 	}
 
+	if !v.cfg.GuardianDisabled {
+		if err := v.guardian.Start(); err != nil {
+			return fmt.Errorf("cannot start guardian: %w", err)
+		}
```

Comment:
When error occurs, we need to stop other components such as `l2os` and `challenger`. Also same to the error in starting of challenger.
```suggestion
		if err := v.guardian.Start(); err != nil {
		        v.cancel()
			return fmt.Errorf("cannot start guardian: %w", err)
		}
```

Reply:
Doesn't the return of an error in the start process induce a cancel via defer at the validator level?